### PR TITLE
Adds Metadata describer

### PR DESCRIPTION
**Pull request by Richard-Gist**
## Checklist:
- [x] Did you sign the [Contributor License Agreement](https://cla-assistant.io/wwt/SwiftCurrent)?
- [x] Is the linter reporting 0 errors? **0 Errors for the code in this PR**
- [ ] Did you comply with our [styleguide](https://github.com/wwt/SwiftCurrent/blob/main/.github/STYLEGUIDE.md)?
- [ ] Is there [adequate test coverage](https://github.com/wwt/SwiftCurrent/blob/main/.github/CONTRIBUTING.md#test-etiquette) for your new code?
- [x] Does the CI pipeline pass? **For what this PR touches**
- [x] Did you [update the documentation](https://github.com/wwt/SwiftCurrent/blob/main/.github/CONTRIBUTING.md#documentation)?
- [ ] Have you done everything you can to make sure that errors that can occur are compile-time errors, and if they have to be runtime do you have adequate tests and documentation around those runtime errors? [For more details](https://github.com/wwt/SwiftCurrent/blob/main/.github/CONTRIBUTING.md#errors).

### CODE REVIEW

File: Sources/SwiftCurrent/Protocols/FlowRepresentable.swift [hunk 1]
```diff
@@ -100,6 +100,16 @@ extension FlowRepresentable {
     // swiftlint:disable:next missing_docs
     public var _workflowUnderlyingInstance: Any { self }
 
+    // False positive: Inherited docs
+    // swiftlint:disable:next missing_docs
+    public static var flowRepresentableName: String { String(describing: Self.self) }
+
+    // False positive: Inherited docs
+    // swiftlint:disable:next missing_docs
+    public static func createMetadata() -> FlowRepresentableMetadata {
```
Comment (Tyler-Keith-Thompson) [inline Sources/SwiftCurrent/Protocols/FlowRepresentable.swift:109]: You may want to match conventions from our `_factory` and call this metadataFactory() or createMetadataFromFactory() or something.

File: Tests/SwiftCurrentTests/FlowRepresentableMetadataDescriberConsumerTests.swift [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+            var _workflowPointer: AnyFlowRepresentable?
+        }
+
+        let FRMD = FR2.self as FlowRepresentableMetadataDescriber.Type
+
+        XCTAssertEqual(FRMD.flowRepresentableName, "FR2")
+        XCTAssertEqual(FRMD.createMetadata().flowRepresentableTypeDescriptor, "FR2")
+    }
+
+    func testExtendingProductsCanProvideUniqueImplementationsForClasses() {
```
Comment (Tyler-Keith-Thompson) [inline Tests/SwiftCurrentTests/FlowRepresentableMetadataDescriberConsumerTests.swift:40]: I'd add an additional test that specifically covers subclasses. If FRSub inherited from FR1 and you asked for its name, for example.

File: Sources/SwiftCurrent/Protocols/FlowRepresentableMetadataDescriber.swift [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    /// The name of the FlowRepresentable as used in the Workflow Data Scheme
+    static var flowRepresentableName: String { get }
+
+    /// Creates a new instance of ``FlowRepresentableMetadata``
+    static func metadataFactory() -> FlowRepresentableMetadata
+}
+
+// Provides the implementation for the protocol without immediately conforming FlowRepresentable
+// See FlowRepresentableMetadataDescriberConsumerTests for reasons.
+extension FlowRepresentable {
```
Comment (Tyler-Keith-Thompson) [inline Sources/SwiftCurrent/Protocols/FlowRepresentableMetadataDescriber.swift:20]: Can you get away with this?

```suggestion
extension FlowRepresentable where Self: FlowRepresentableMetadataDescriber {
```

**Comment by Richard-Gist** [comment]
I think this is probably good enough to start updating this to Ready for Review.  I'm out for most if not all of next week (and tomorrow) so if we want to make more progress on this, feel free to take it and keep working on it without me.

**Review by morganzellers — APPROVED** [review]

**Comment by morganzellers** [comment]
Looks like the linter is only failing due to two TODOs that I feel are appropriate to stay until `data-driven` merges into `main`, so I'm approving & merging.

**Pull request merged**